public/app/themes/justice/src/patch/js/ccfw-gtm.js

```diff
@@ -57,167 +69,189 @@ const CCFW = {
             } else {
                 if (allowList.indexOf(allowed) === -1) {
-                    allowList.push(allowed);
+                    allowList.push(allowed)
                 }
-                jQuery(element).attr('aria-checked', true);
-                jQuery('#ccfw-' + allowed + '-toggle-on').removeAttr('aria-hidden').show();
-                jQuery('#ccfw-' + allowed + '-toggle-off').attr('aria-hidden', 'true').hide();
+                jQuery(element).attr('aria-checked', true)
+                jQuery('#ccfw-' + allowed + '-toggle-on').removeAttr('aria-hidden').show()
+                jQuery('#ccfw-' + allowed + '-toggle-off').attr('aria-hidden', 'true').hide()
             }
-        });
+        })
 
-        return allowList;
+        return allowList
     },
+    cache: () => CCFW.jq.toggles.each((key, element) => CCFW.allowedIds.push(jQuery(element).data('allowlist'))),
     manageAll: (allowList, allowed, pressed) => {
         if (!allowList) {
-            return;
+            return
         }
 
-        let totalAllowed = jQuery('.' + CCFW.selector.toggles).length - 1;
+        let totalAllowed = CCFW.jq.toggles.length - 1
 
         if (allowed !== 'all') {
             if (pressed) {
-                allowList = allowList.filter(item => item !== 'all');
-                jQuery('#ccfw-all-toggle-off').removeAttr('aria-hidden').show();
-                jQuery('#ccfw-all-toggle-on').attr('aria-hidden', 'true').hide();
+                allowList = allowList.filter(item => item !== 'all')
+                jQuery('#ccfw-all-toggle-off').removeAttr('aria-hidden').show()
+                jQuery('#ccfw-all-toggle-on').attr('aria-hidden', 'true').hide()
                 if (allowList.length === 0) {
-                    jQuery('button[data-allowlist="all"]').attr('aria-checked', false);
+                    jQuery('button[data-allowlist="all"]').attr('aria-checked', false)
                 }
             }
 
             if (allowList.length > 0) {
-                jQuery('button[data-allowlist="all"]').attr('aria-checked', true);
+                jQuery('button[data-allowlist="all"]').attr('aria-checked', true)
             }
 
             if (totalAllowed === allowList.length) {
                 if (allowList.indexOf('all') === -1) {
-                    allowList.push('all');
+                    allowList.push('all')
                 }
-                jQuery('#ccfw-all-toggle-on').removeAttr('aria-hidden').show();
-                jQuery('#ccfw-all-toggle-off').attr('aria-hidden', 'true').hide();
-                jQuery('button[data-allowlist="all"]').attr('aria-checked', true);
+                jQuery('#ccfw-all-toggle-on').removeAttr('aria-hidden').show()
+                jQuery('#ccfw-all-toggle-off').attr('aria-hidden', 'true').hide()
+                jQuery('button[data-allowlist="all"]').attr('aria-checked', true)
             }
         }
 
-        return allowList;
+        return allowList
+    },
+    hack: {
```

Comment:
Could we rename this it we're still thinking about pushing the work upstram?

Reply:
Sure 👍🏼